js/src/hooks/useActiveIssueType.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * External dependencies
+ */
+import { getQuery } from '@woocommerce/navigation';
+
+/**
+ * Internal dependencies
+ */
+import { ISSUE_TYPE_ACCOUNT, ISSUE_TYPE_PRODUCT } from '.~/constants';
+import useMCIssuesTotals from '.~/hooks/useMCIssuesTotals';
+
+/**
+ * Gets the active Issue Type Filter based on the `issueType` query property.
+ *
+ * @return {string} The active Issue Type Filter based on the url query. If `issueType` is falsy because is not in the query
+ * it returns the issue type account if it has any issue, or it returns issue type product otherwise.
+ */
+const useActiveIssueType = () => {
+	const issueTotals = useMCIssuesTotals();
+
+	const defaultIssueType = issueTotals[ ISSUE_TYPE_ACCOUNT ]
+		? ISSUE_TYPE_ACCOUNT
+		: ISSUE_TYPE_PRODUCT;
```

Comment:
🚧  This method will cause users to be unable to back to the previous page by browsers navigate button.

https://user-images.githubusercontent.com/17420811/150917144-15ae5632-79fd-41fe-9fa4-858a402224d9.mp4

Replicate steps:
1. Replace `getMCIssues()` in resolvers.js with the following:
    ```js
    import { __unstableAwaitPromise } from '@wordpress/data-controls';
    
    export function* getMCIssues( query ) {
    	const response = { page: 1, total: 0, issues: [] };
    
    	if ( query.issue_type === 'product' ) {
    		yield __unstableAwaitPromise(
    			new Promise( ( resolve ) => setTimeout( resolve, 200 ) )
    		);
    		response.total = 1;
    		response.issues.push( {
    			type: 'product',
    			action: 'Use a complete URL',
    			action_url: 'https://support.google.com/merchants/answer/7052112',
    			issue: 'Invalid URL',
    			product: 'Cap',
    			severity: 'error',
    		} );
    	}
    
    	yield receiveMCIssues( query, response );
    }
    ```
2. Go to Dashboard page.
3. Refresh page.
4. Go to Product Feed.
5. After issues table is loaded, it's unable to back to the previous page by browser navigate button.

---

What do you think if we replace this condition with `if ( ! totals.total )` instead of updating URL query in the hook?

https://github.com/woocommerce/google-listings-and-ads/blob/c195abd493fe27eb2f5a3e970a8f294d041a8c6d/js/src/product-feed/issues-table-card/index.js#L58-L59

Reply:
Again, thanks for catching this :)

I did a new fix here https://github.com/woocommerce/google-listings-and-ads/pull/1196/commits/a837decfd965872f424d1b717d18d1fe6373cd3f